test/mpi/main.cc

```diff
@@ -596,50 +610,51 @@ int main(int argc, char *argv[])
             }
         }
     }
-
     std::cout << std::flush;
-    MPI_Iallreduce(MPI_IN_PLACE, test->results.data(), test->results.size(),
-                   MPI_INT, MPI_MIN, MPI_COMM_WORLD, &req);
+
+    total_done_skipped_failed[0] = test->results.size();
+    for (auto s : test->results) {
+        switch(s) {
+        case UCC_OK:
+            total_done_skipped_failed[1]++;
+            break;
+        case UCC_ERR_NOT_IMPLEMENTED:
+        case UCC_ERR_LAST:
+            total_done_skipped_failed[2]++;
+            break;
+        default:
+            total_done_skipped_failed[3]++;
+        }
+    }
+    MPI_Iallreduce(MPI_IN_PLACE, total_done_skipped_failed,
+                   sizeof(total_done_skipped_failed)/sizeof(int),
+                   MPI_INT, MPI_MAX, MPI_COMM_WORLD, &req);
```

Comment:
why not just a "reduce" rooted in rank=0?

Reply:
essentially will have the same effect, it was allreduce here before so I keep it as allreduce. Can replace with reduce if you think it's important